src/wallet/wallet.cpp

```diff
@@ -2395,15 +2395,37 @@ bool CWallet::DelAddressBook(const CTxDestination& address)
             WalletLogPrintf("%s called with IsMine address, NOT SUPPORTED. Please report this bug! %s\n", __func__, PACKAGE_BUGREPORT);
             return false;
         }
+
+        WalletBatch batch(GetDatabase());
+        if (!batch.TxnBegin()) {
+            WalletLogPrintf("Error: cannot create db txn for address book entry removal\n");
+            return false;
+        }
+
         // Delete data rows associated with this address
-        batch.EraseAddressData(address);
+        if (!batch.EraseAddressData(address)) {
+            WalletLogPrintf("Error: cannot erase address book entry data\n");
+            return false;
+        }
+
+        // Delete purpose and name
+        if (!batch.ErasePurpose(dest) || !batch.EraseName(dest)) {
+            WalletLogPrintf("%s error erasing purpose/name\n", __func__);
```

Comment:
> Any reason to add `__func__` to this log, when none of the other logs have it here?

Not really. It is just a bad habit. I pulled 9a4e1f5 from #26836 because it is required for this changes. Will change it there.